polaris-react/src/components/DropZone/DropZone.stories.tsx

```diff
@@ -424,8 +424,155 @@ export function SmallSized() {
   );
 }
 
+export function SmallSizedWithCustomFilePicker() {
+  const [activeModal, setActiveModal] = useState(false);
+  const [files, setFiles] = useState<File[]>([]);
+  const [rejectedFiles, setRejectedFiles] = useState<File[]>([]);
+  const hasError = rejectedFiles.length > 0;
+
+  const handleThumbnailDrop = useCallback(
+    (_droppedFiles: File[], acceptedFiles: File[], rejectedFiles: File[]) => {
+      setFiles((files) => [...files, acceptedFiles[0]]);
+      setRejectedFiles(rejectedFiles);
+    },
+    [],
+  );
+
+  const handleModalToggle = () => setActiveModal((activeModal) => !activeModal);
+
+  const handleKeypress = (event: React.KeyboardEvent) => {
+    event.stopPropagation();
+    if (event.key === 'Enter' || event.key === 'Space') {
+      handleModalToggle();
+    }
+  };
+
+  const thumbnailStyle = {
+    background: 'none',
+    border: 'none',
+    padding: 0,
+    borderRadius: 'var(--p-border-radius-200)',
+    outlineOffset: 'var(--p-border-width-050)',
+    cursor: 'pointer',
+    overflow: 'hidden',
+    minHeight: '0',
+    minWidth: '0',
+    display: 'flex',
+  };
+
+  const fileUploadThumbnail = (
+    <div style={{width: '40px', height: '40px'}}>
+      <DropZone
+        dragAndDropOnly
+        error={hasError}
+        allowMultiple={false}
+        onClick={handleModalToggle}
+        onDrop={handleThumbnailDrop}
+      >
+        <UnstyledButton style={thumbnailStyle} onKeyUp={handleKeypress}>
```

Comment:
The interactive child should probably be configured to handle click event instead of the DropZone's wrapper div)